### PR TITLE
Separate store types

### PR DESCRIPTION
**Pull request by omid**
Here I separated the store types to make it clearer and easier to spot issues. 

I had to:
- Repeat the test methods, which is not bad :)
- Make some cache store properties, `pub(super)`, but for consistency, I made all of them `pub(super)`!
- Create some type aliases in the super module to be backward compatible.

Now, instead of having a ~2K file, we have 5 files between 250-650 lines.

PS: This PR is so opinionated, feel free to reject it :)

### CODE REVIEW

**Comment by jaemk** [comment]
Nice, this was sorely needed. Unfortunately there was a separate PR that had gone through a couple conflict resolution cycles so I merged that one first. Would you mind resolving those conflicts to the `SizedCache` store? Probably easiest to just copy/paste the whole thing again into the new file

**Comment by omid** [comment]
@jaemk sure, no problem.

File: src/stores/mod.rs [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+
+#[cfg(feature = "async")]
+use {super::CachedAsync, async_trait::async_trait, futures::Future};
+
+pub mod sized;
+pub mod timed;
+pub mod timed_sized;
+pub mod unbound;
+
+pub type UnboundCache<K, V> = unbound::UnboundCache<K, V>;
```
Comment (jaemk) [inline src/stores/mod.rs:16]: do these need to be defined as `pub type` or could they be a `pub use unbound::UnboundCache`, etc?

File: src/stores/mod.rs [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+pub type SizedCache<K, V> = sized::SizedCache<K, V>;
+pub type TimedSizedCache<K, V> = timed_sized::TimedSizedCache<K, V>;
+pub type TimedCache<K, V> = timed::TimedCache<K, V>;
+
+/// Limited functionality doubly linked list using Vec as storage.
+#[derive(Clone, Debug)]
+struct LRUList<T> {
```
Comment (jaemk) [inline src/stores/mod.rs:23]: could we move this to a `pub(crate) mod lru_list` to clear out this file while we're at it?

**Comment by jaemk** [comment]
Looks good @omid , just two comments/questions

File: src/stores/mod.rs (hunk 2 above)
Comment (omid) [reply src/stores/mod.rs:23]: My plan was to keep everything inside the mod.rs file, except the public store types.
But that's fine, so I will put it next to the `lib.rs`.

Any preference for `impl<K: Hash + Eq, V> Cached<K, V> for HashMap<K, V>`?

File: src/stores/mod.rs (hunk 1 above)
Comment (omid) [reply src/stores/mod.rs]: Done. Also `pub mod unbound;` and all others should not be public!

**Comment by omid** [comment]
@jaemk How can I push it forward? :)

**Comment by jaemk** [comment]
Sorry, fell off my radar - Looks good! Thanks @omid !

**Pull request merged**